includes/Linker.php

```diff
@@ -1796,11 +1796,15 @@ public static function generateRollback( $rev ) {
 	public static function buildRollbackLink( $rev ) {
 		global $wgRequest, $wgUser;
 		$title = $rev->getTitle();
-		$query = array(
+		// SUS-812: use user name lookup to determine link target
+		$userName = User::getUsername( $rev->getUser(), $rev->getUserText() );
+
+		$query = [
 			'action' => 'rollback',
-			'from' => $rev->getUserText(),
-			'token' => $wgUser->getEditToken( [ $title->getPrefixedText(), $rev->getUserText() ] ),
-		);
+			'from' => $userName,
+			'token' => $wgUser->getEditToken( [ $title->getPrefixedText(), $userName ] ),
+		];
+
 		if ( $wgRequest->getBool( 'bot' ) ) {
 			$query['bot'] = '1';
 			$query['hidediff'] = '1'; // bug 15999
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app